Modules/posixmodule.c

```diff
@@ -8263,32 +8244,39 @@ os_setpgrp_impl(PyObject *module)
 #ifdef HAVE_GETPPID
 
 #ifdef MS_WINDOWS
-#include <tlhelp32.h>
+#include <processsnapshot.h>
 
 static PyObject*
 win32_getppid()
 {
-    HANDLE snapshot;
+    PSS_THREAD_ENTRY thread;
     PyObject* result = NULL;
-    BOOL have_record;
-    PROCESSENTRY32 pe;
-
+    HPSS snapshot = NULL;
+    HPSSWALK walk = NULL;
     DWORD mypid = GetCurrentProcessId(); /* This function never fails */
+    HANDLE myhandle = GetCurrentProcess();
 
-    snapshot = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
-    if (snapshot == INVALID_HANDLE_VALUE)
-        return PyErr_SetFromWindowsErr(GetLastError());
+    if (PssCaptureSnapshot(myhandle, PSS_CAPTURE_THREADS, 0, &snapshot) != ERROR_SUCCESS)
```

Comment:
I didn't expect you to implement this. It was just an offhand comment. Also, I was mistakenly looking at the Windows 11 SDK when I made the comment. In Windows 10, the process snapshot API is limited to the desktop and system API partitions. They expanded it to support the app API partition in Windows 11.

FWIW, the implementation is actually much simpler than you're making it. Call `PssCaptureSnapshot(GetCurrentProcess(), PSS_CAPTURE_NONE, 0, &hsnapshot)`, and query `PssQuerySnapshot(hsnapshot, PSS_QUERY_PROCESS_INFORMATION, &info, sizeof(info))`. The `info` buffer is a `PSS_PROCESS_INFORMATION` record. The parent process ID is its `ParentProcessId` field.

Reply:
Yes this is indeed a lot simpler + apparently my implementation was not even correct :see_no_evil: